pkg/cluster/sync.go

```diff
@@ -568,8 +568,17 @@ func (c *Cluster) checkAndSetGlobalPostgreSQLConfiguration(pod *v1.Pod, effectiv
 		configToSet["ttl"] = desiredPatroniConfig.TTL
 	}
 
+	slotsToSet := make(map[string]interface{})
+	// check if there is any slot deletion
+	for slotName, effectiveSlot := range effectivePatroniConfig.Slots {
+		if desiredSlot, exists := desiredPatroniConfig.Slots[slotName]; exists {
+			if reflect.DeepEqual(effectiveSlot, desiredSlot) {
+				continue
+			}
+		}
+		slotsToSet[slotName] = nil
```

Comment:
this will also remove slots that were already existing in Postgres/Patroni config but not in the manifest, right? We should only remove slots when they were specified in the manifest before. I think we would need to cache all slots from oldSpec on UPDATE events and only do `slotsToSet[slotName] = nil` when slotName exists in that cache.